test/e2e/main_test.go

```diff
@@ -3,9 +3,77 @@ package e2e
 import (
 	"testing"
 
-	f "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/keycloak/keycloak-operator/pkg/apis"
+	keycloakv1alpha1 "github.com/keycloak/keycloak-operator/pkg/apis/keycloak/v1alpha1"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
 )
 
+type deployedOperatorTestStep func(*testing.T, *framework.Framework, *framework.TestCtx, string) error
+type environmentInitializationStep func(*testing.T, *framework.Framework, *framework.TestCtx, string) error
+
+type CRDTest interface {
+	PrepareEnvironment() []environmentInitializationStep
+
+	GetSubsetOfTests() map[string]deployedOperatorTestStep
+}
+
 func TestMain(m *testing.M) {
-	f.MainEntry(m)
+	framework.MainEntry(m)
+
+}
+
+func TestKeycloakCRDS(t *testing.T) {
+	keycloakType := &keycloakv1alpha1.Keycloak{}
+	err := framework.AddToFrameworkScheme(apis.AddToScheme, keycloakType)
+	if err != nil {
+		t.Fatalf("failed to add custom resource scheme to framework: %v", err)
+	}
+
+	t.Run("KeycloaksCRDTest", func(t *testing.T) {
+		runTestsFromCRDInterface(t, &KeycloaksCRDTestStruct{})
+	})
+	t.Run("KeycloakBackupCRDTest", func(t *testing.T) {
+		runTestsFromCRDInterface(t, &KeycloakBackupCRDTestStruct{})
+	})
+}
+
+func runTestsFromCRDInterface(t *testing.T, crd CRDTest) {
+
+	for testName, testMethod := range crd.GetSubsetOfTests() {
```

Comment:
In Go, you can just take values from a struct. There is no need to implement accessors like in Java.